src/main/java/com/haxwell/apps/quizki/config/WebSecurityConfig.java

```diff
@@ -54,6 +54,8 @@ protected void configure(HttpSecurity http) throws Exception {
         	.authorizeRequests()
 	    		.antMatchers(HttpMethod.POST, "/api/users").permitAll()
 	    		// TODO: Handle login.. That should be open.
+	    		.antMatchers(HttpMethod.GET, "/api/verifyCredentials").permitAll()
```

Comment:
We want verifyCredentials to be inaccessible if you don't have a username and password.. See the comments in VerifyCredentialsController.java.

Reply:
I was getting an `Access Denied` error from the frontend request with the basic authentication header. It seemed like a catch 22, to log in you needed to be logged in. Not sure how to change this. I think there would always need to be a publicly accessible endpoint to start the login process and so I thought this would be it.

Reply:
In my testing, I ran into this problem too.. The solution was to set the `HttpMethod.OPTIONS` request for `/api/verifyCredentials` to `permitAll()`. `GET` remains authenticated.